crates/store/re_chunk/src/iter.rs

```diff
@@ -291,6 +298,69 @@ impl Chunk {
                 }),
         )
     }
+
+    /// Returns an iterator over the raw buffers of a [`Chunk`], for a given component.
+    ///
+    /// This is a very fast path: the entire column will be downcasted at once, and then every
+    /// component batch will be a slice reference into that global slice.
+    /// Use this when working with simple arrow datatypes and performance matters (e.g. blobs, etc).
+    ///
+    /// See also:
+    /// * [`Self::iter_primitive`]
+    /// * [`Self::iter_primitive_array`]
+    /// * [`Self::iter_string`].
+    /// * [`Self::iter_component_arrays`].
+    /// * [`Self::iter_component`].
+    pub fn iter_buffer<T: arrow2::types::NativeType>(
+        &self,
+        component_name: &ComponentName,
+    ) -> impl Iterator<Item = Vec<ArrowBuffer<T>>> + '_ {
+        let Some(list_array) = self.components.get(component_name) else {
+            return Either::Left(std::iter::empty());
+        };
+
+        let Some(inner_list_array) = list_array
+            .values()
+            .as_any()
+            .downcast_ref::<ArrowListArray<i32>>()
+        else {
+            if cfg!(debug_assertions) {
+                panic!("downcast failed for {component_name}, data discarded");
+            } else {
+                re_log::error_once!("downcast failed for {component_name}, data discarded");
+            }
+            return Either::Left(std::iter::empty());
+        };
+
+        let Some(values) = inner_list_array
+            .values()
+            .as_any()
+            .downcast_ref::<ArrowPrimitiveArray<T>>()
+        else {
+            if cfg!(debug_assertions) {
+                panic!("downcast failed for {component_name}, data discarded");
+            } else {
+                re_log::error_once!("downcast failed for {component_name}, data discarded");
+            }
+            return Either::Left(std::iter::empty());
+        };
+
+        let values = values.values();
+        let offsets = inner_list_array.offsets();
+        let lengths = inner_list_array.offsets().lengths().collect_vec();
+
+        // NOTE: No need for validity checks here, `iter_offsets` already takes care of that.
+        Either::Right(
+            self.iter_component_offsets(component_name)
+                .map(move |(idx, len)| {
+                    let offsets = &offsets.as_slice()[idx..idx + len];
+                    let lengths = &lengths.as_slice()[idx..idx + len];
+                    izip!(offsets, lengths)
+                        .map(|(&idx, &len)| values.clone().sliced(idx as _, len).into())
```

Comment:
that clone here is a ref count based clone of arrow buffers, correct? :)

Reply:
Indeed, I'll add my usual lil' comment